flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/factories/TestValuesRuntimeFunctions.java

```diff
@@ -378,6 +395,8 @@ public void invoke(RowData value, Context context) throws Exception {
                 }
                 synchronized (LOCK) {
                     localRawResult.add((Row) converter.toExternal(value));
+                    notifyRawResultsObservers(
```

Comment:
method is called 4 times with the same args? maybe create a method in super class?